src/Core/Addon.php

```diff
@@ -0,0 +1,499 @@
+			}
+		}
+	}
+
+	/**
+	 * @brief Calls a single hook.
+	 *
+	 * @param string $name of the hook to call
+	 * @param array $hook Hook data
+	 * @param string|array &$data to transmit to the callback handler
+	 */
+	public static function callSingleHook($a, $name, $hook, &$data = null)
+	{
+		// Don't run a theme's hook if the user isn't using the theme
+		if (strpos($hook[0], 'view/theme/') !== false && strpos($hook[0], 'view/theme/'.current_theme()) === false) {
+			return;
+		}
+
+		@include_once($hook[0]);
+		if (function_exists($hook[1])) {
+			$func = $hook[1];
+			$func($a, $data);
+		} else {
+			// remove orphan hooks
+			$condition = ['hook' => $name, 'file' => $hook[0], 'function' => $hook[1]];
+			dba::delete('hook', $condition);
+		}
+	}
+
+	/**
+	 * Check if an app_menu hook exists for plugin
+	 *
+	 * @param string $name app name
+	 * @return boolean
+	 */
+	public static function isApp($name)
+	{
+		$a = get_app();
+
+		if (is_array($a->hooks) && (array_key_exists('app_menu', $a->hooks))) {
+			foreach ($a->hooks['app_menu'] as $hook) {
+				if ($hook[0] == 'addon/'.$name.'/'.$name.'.php') {
+					return true;
+				}
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Parse plugin comment in search of plugin infos.
+	 *
+	 * like
+	 * \code
+	 *...* Name: Plugin
+	*   * Description: A plugin which plugs in
+	* . * Version: 1.2.3
+	*   * Author: John <profile url>
+	*   * Author: Jane <email>
+	*   *
+	*  *\endcode
+	* @param string $plugin the name of the plugin
+	* @return array with the plugin information
+	*/
+	public static function getPluginInfo($plugin)
+	{
+		$a = get_app();
+
+		$info=[
+			'name' => $plugin,
+			'description' => "",
+			'author' => [],
+			'version' => "",
+			'status' => ""
+		];
+
+		if (!is_file("addon/$plugin/$plugin.php")) {
+			return $info;
+		}
+
+		$stamp1 = microtime(true);
+		$f = file_get_contents("addon/$plugin/$plugin.php");
+		$a->save_timestamp($stamp1, "file");
+
+		$r = preg_match("|/\*.*\*/|msU", $f, $m);
+
+		if ($r) {
+			$ll = explode("\n", $m[0]);
+			foreach ($ll as $l) {
+				$l = trim($l, "\t\n\r */");
+				if ($l != "") {
+					list($k,$v) = array_map("trim", explode(":", $l, 2));
+					$k= strtolower($k);
+					if ($k == "author") {
+						$r=preg_match("|([^<]+)<([^>]+)>|", $v, $m);
+						if ($r) {
+							$info['author'][] = ['name'=>$m[1], 'link'=>$m[2]];
+						} else {
+							$info['author'][] = ['name'=>$v];
+						}
+					} else {
+						if (array_key_exists($k, $info)) {
+							$info[$k]=$v;
+						}
+					}
+				}
+			}
+		}
+		return $info;
+	}
+
+
+	/**
+	 * Parse theme comment in search of theme infos.
+	 *
+	 * like
+	 * \code
+	 * ..* Name: My Theme
+	 *   * Description: My Cool Theme
+	 * . * Version: 1.2.3
+	 *   * Author: John <profile url>
+	 *   * Maintainer: Jane <profile url>
+	 *   *
+	 * \endcode
+	 * @param string $theme the name of the theme
+	 * @return array
+	 */
+	public static function getThemeInfo($theme)
```

Comment:
I would prefer having the theme stuff in another file.

Reply:
Please move to `Friendica\Core\Theme::getInfo()`.